### ISSUE TITLE
Expose `block-bg-padding` SCSS variables as CSS variables

### ISSUE THREAD

**Issue by Luehrsen** [WordPress/gutenberg#51323]
## What problem does this address?

Some core blocks generate padding around the content when a background color is added. This padding is typically triggered by the `.has-background` class, with padding values defined in SCSS as `$block-bg-padding--v` and `$block-bg-padding--h`.

Currently, these values are hardcoded in SCSS and aren't exposed as CSS variables. This situation forces developers who are extending the functionality with custom blocks to also hardcode these values when they aim to replicate the `.has-background` behavior. 

See:
https://github.com/WordPress/gutenberg/blob/5e1a0a8deadd28b747bffdddb4b576bfbac72829/packages/base-styles/_variables.scss#L103-L105

https://github.com/WordPress/gutenberg/blob/3cef94b4afe86610dae54ace24a9057ca97a7618/packages/block-library/src/paragraph/style.scss#L41-L43

https://github.com/WordPress/gutenberg/blob/bdf728b7bf7105ad7ee576200aa46dc82ff424fb/packages/block-library/src/heading/style.scss#L1-L10

## What is your proposed solution?

The suggestion is to expose these two variables as CSS variables. This change would effectively future-proof the adjustment of these padding values. For example, there are a few discussions about modifying these values to more sensible alternatives, such as `blockGap` (@richtabor).

Additionally, exposing these values as CSS variables would allow themes to easily customize these values. If implemented appropriately, this solution would provide extenders with a dependable method of emulating the `.has-background` functionality (similar to the paragraph block) without needing to hardcode values. This would also enable these extensions to be seamlessly integrated into potential theme changes.

---

(This issue was written at WCEU Contributor Day 2023 on the Core table)

**Comment by ndiego** [comment]
I believe this practice of auto-adding padding when a background color is set began when we didn't have dimension controls on most blocks. Personally, I wish that no padding was added by default because it feels quite arbitrary. Some blocks have it, some don't, and the padding that is added has no relation to the dimension sizes set in a theme's theme.json file. But I believe that removing this would cause backward compatibility issues. 🤔

**Comment by Luehrsen** [comment]
> I believe that removing this would cause backward compatibility issues. 🤔

Yes. And I actually like if the editor and/or the theme can provide sensible defaults, that can easily overridden on user choice. Because removing this would create a 'default broken' layout with the background colors on select and would force the user to customize another value to make it work.

Edit: 

For clarification, this is what I would consider 'default broken':
<img width="662" alt="Bildschirmfoto 2023-06-08 um 15 34 51" src="https://github.com/WordPress/gutenberg/assets/1105871/8e29725c-04d5-4aba-8219-c776abd72e64">

When you remove the practice of auto-adding padding with `.has-background`, users will always see this non-desireable state first, before they have to add sensible paddings themselves.

**Comment by aurooba** [comment]
Agreed! Removing the padding would be a worse experience, but connecting this auto-padding to theme.json and/or exposing that CSS Variable would be _very_ helpful ++++1

**Comment by ndiego** [comment]
> Agreed! Removing the padding would be a worse experience, but connecting this auto-padding to theme.json and/or exposing that CSS Variable would be very helpful ++++1

I think it depends on the user. I manually suppress this auto padding in all my work 😅 Regardless, a consistent way to apply/modify/remove this padding would be great.
